xml2rfc_compat/serializers/reference.py

```diff
@@ -106,19 +106,26 @@ def create_reference(item: BibliographicItem) -> _Element:
             func(docid)
             for func in DOCID_SERIES_EXTRACTORS
         ])
+
+    doi = False
     for series_info in list(dict.fromkeys(series)):
         if series_info is not None:
             ref.append(E.seriesInfo(
                 name=series_info[0],
                 value=series_info[1],
             ))
+            if series_info[0] == "DOI":
+                doi = True
 
     # Target, may be overwritten by callers
     try:
         target = get_suitable_target(as_list(item.link or []))
     except ValueError:
         pass
     else:
+        if doi:
+            # https://github.com/ietf-tools/bibxml-service/issues/332
+            target = target.replace("http", "https").replace("dx.doi.org", "doi.org")
```

Comment:
This is a bit dangerous as `http` (and, less likely, `dx.doi.org`) could occur in the actual doi.  I would probably replace `http:` with `https:` and `//dx.doi.org` with `//doi.org`.

Reply:
Good catch @cabo.
@stefanomunarini, Maybe using [Python URL Parse](https://docs.python.org/3/library/urllib.parse.html#url-parsing) will make things clear?